timer/component.go

```diff
@@ -0,0 +1,34 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package timer
+
+import "sync/atomic"
+
+type component struct {
+	// name through which the component is identified
+	name string
+	// Updated each time a run of a sub-timer ends
+	spent atomic.Int64
+}
+
+// components store the data shared between child timers of the same component name
+type components struct {
+	lookup  map[string]*component
+	storage []component
+}
+
+func newComponents(names []string) components {
+	lookup := make(map[string]*component, len(names))
+	storage := make([]component, len(names))
+	for i, name := range names {
+		storage[i].name = name
+		lookup[name] = &storage[i]
+	}
+	return components{
+		lookup:  lookup,
+		storage: storage,
+	}
+}
```

Comment:
Considering this usage, what's the added value of `storage`?

Feels like `components` could be `type components map[string]component`, and it'd have the same semantics, modulo ability to read the components in deterministic creation order; while incurring 1 allocation fewer?

Reply:
What you are describing does not work. doing a call to `lookup[name]` would copy the atomic around I would not be able to sum anything. If I do what you asked I have to have a pointer pointing to the atomic somewhere. Moreover benchmarks tell me that doing only a call to `make` is better performance-wise than doing 5 or 10 calls to `new`. 

But this made me think and I trying do simplify it a bit further.